mlir/lib/Dialect/MQTOpt/Transforms/Transpilation/sc/PlacementPass.cpp

```diff
@@ -447,14 +459,23 @@ struct PlacementPassSC final : impl::PlacementPassSCBase<PlacementPassSC> {
       std::random_device rd;
       const std::size_t seed = rd();
       LLVM_DEBUG({
-        llvm::dbgs() << "getPlacer: random placement with seed = " << seed
+        llvm::dbgs() << "getPlacer: random placement with seed =" << seed
                      << '\n';
       });
       return std::make_unique<RandomPlacer>(arch.nqubits(),
                                             std::mt19937_64(seed));
     }
     llvm_unreachable("Unknown strategy");
   }
+
+  LogicalResult preflight() {
+    if (archName.empty()) {
+      return emitError(UnknownLoc::get(&getContext()),
+                       "required option 'arch' not provided");
+    }
+
+    return success();
+  }
```

Comment:
I am surprised that it is not supported out of the box to mark a parameter of a pass as required. Probably worth a look.